DM-30162: Update multiband-anlysis for new deblended flags (#176)
lsst/pipelines_lsst_io: getting-started/multiband-analysis.rst

Exactly what filtering is done depends on the application, but source tables should *always* be filtered for unique sources.
There are two ways that measured sources might not be unique: deblended sources and sources in patch overlaps.
Additionally, some sources are "sky" objects added by ``detectCoaddSources.py`` for noise characterization that you need to filter out.

> This section gives a brief introduction to removing duplicate and unwanted sources, but a more thorough description can be found in `pipe_tasks <https://pipelines.lsst.io/modules/lsst.pipe.tasks/index.html>`_.

Reviewer: ```suggestion
This section gives a brief introduction to removing duplicate and unwanted sources, for details see :doc:`/modules/lsst.pipe.tasks/deblending-flags-overview`.
```

Reasons:

- This creates a relative link within a documentation build rather than an external link that doesn't work with versioned documentation.
- Automatically inserts the title of the destination page.
- Points to the deblending flags page rather than the homepage for `lsst.pipe.tasks`.

> This section gives a brief introduction to removing duplicate and unwanted sources, but a more thorough description can be found in `this Community post <https://community.lsst.org/t/new-use-of-isprimary-that-began-in-w-2021-12/4957>`_.

Reviewer: It's great that you wrote that Community post, but keep in mind that our documentation doesn't live on the Community forum. The forum links to pipelines.lsst.io, but not the other way around. I'd like us to delete this link to community.lsst.org and instead link to a page in the `lsst.pipe.tasks` documentation.

You can start by literally copying the content from the [forum post](https://community.lsst.org/t/new-use-of-isprimary-that-began-in-w-2021-12/4957) and adding a new rst file to `pipe_tasks` at `doc/lsst.pipe.tasks/deblending-flags-overview.rst` (and add that [file to the "Using lsst.pipe.tasks" toctree](https://developer.lsst.io/stack/module-homepage-topic-type.html#using-module-and-toctrees) in doc/lsst.pipe.tasks/index.rst).

This is done so that you can choose whether to use blended or deblended measurements in your analysis.
If you *don't* choose, though, the same flux will be included multiple times in your analysis.

> Usually you will want to use fully-deblended sources in your analysis, however there are two different types of deblended sources: isolated sources that were modeled by scarlet (currently used mostly for diagnostic purposes) and children of parents that contain multiple children.

Reviewer: ```suggestion
Usually you will want to use fully-deblended sources in your analysis, however there are two different types of deblended sources: isolated sources that were modeled by scarlet (used mostly for diagnostic purposes) and children of parents that contain multiple children.
```

It's a good idea to not hedge against the future in documentation.

> You will most likely want the catalog of un-modeled isolated sources and scarlet models for children of blends, which can be found using

Reviewer: ```suggestion
You will most likely want the catalog of un-modeled isolated sources and scarlet models for children of blends:
```

We avoid incomplete sentences when introducing code samples, lists, etc.

.. code-block:: python

   isDeblended = rSources['detect_isDeblendedSource']

> But if you're worried about an inconsistency between the models for isolated sources and blended sources you can instead use the scarlet models for all deblended sources using

Reviewer: ```suggestion
But if you're worried about an inconsistency between the models for isolated sources and blended sources you can instead use the scarlet models for all deblended sources:
```
